### DISCUSSION TITLE
Linking MadAnalysis - ROOT - Delphes for Apple M1 chip

### DISCUSSION THREAD

**Discussion by Altakach313** [MadAnalysis/madanalysis5#173] [General]
I will share here the versions of ma5, root and delphes that were compatible and worked on my Mac and how I installed them.

I first started by cloning ma5:
```
git clone https://github.com/MadAnalysis/madanalysis5.git
MA5 release : 1.10.7 [ 2023/01/06 ]
```

Then from within ma5 I installed zlib and fastjet:
```
install zlib
install fastjet
```
Then I went to `~/madanalysis5/tools` and I downloaded `root_v6.26.10.macos-12.6-arm64-clang140.tar`, this version is compatible with the ARM architecture and can be found [here](https://root.cern/releases/release-62610/).
It is a precompiled version so it is enough to untar.
In order to check if root is well installed one need to:
```
cd root
source bin/thisroot.sh
root
```
At this stage you will get something like that: macOS cannot verify that this app is free from malware. And one can handle it as usual from system preferences. But this message will pop out for so so so many root libraries. In order to allow all of them one need to write the following in the Terminal:
```sudo spctl --master-disable```
(type: `sudo spctl --master-enable` , if you want to go back to the previous state)

Now start ma5 (`./bin/ma5`) and you should see a green ok next to root.

The next step is to install delphes. I went back to `/madanalysis5/tools` and cloned it (version 3.5.0):
```git clone https://github.com/delphes/delphes.git```
then cd delphes and hit make.

Now when `./bin/ma5` one should see a green ok near Delphes.

At this stage everything should work but one might get the following error message:
```
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootTreeReader.h
   requested to autoload type ExRootTreeReader
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootTreeWriter.h
   requested to autoload type ExRootTreeWriter
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootTreeBranch.h
   requested to autoload type ExRootTreeBranch
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootResult.h
   requested to autoload type ExRootResult
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootClassifier.h
   requested to autoload type ExRootClassifier
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootFilter.h
   requested to autoload type ExRootFilter
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootProgressBar.h
   requested to autoload type ExRootProgressBar
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootConfReader.h
   requested to autoload type ExRootConfParam
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootConfReader.h
   requested to autoload type ExRootConfReader
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
   Missing FileEntry for ExRootAnalysis/ExRootTask.h
   requested to autoload type ExRootTask
```
In order to fix this, one need to:

```export ROOT_INCLUDE_PATH=/path/to/madanalysis5/tools/delphes/external```

**Comment by jackaraz** [comment]
Thanks a lot for this comment @Altakach313; for the future, I suggest using [releases](https://github.com/MadAnalysis/madanalysis5/releases) instead of cloning the repository. Our `main` branch is constantly under development; hence it's not the stable version. We only release when the package is widely tested.

PS: @BenjaminFuks independently confirmed that your suggestion works!

**Comment by BFuks** [comment]
Thanks a lot @Altakach313!  I have created [this PR](https://github.com/MadAnalysis/madanalysis5/pull/174) so that the proposed changes are now included in v1.10.9. This restores compatibility with Delphes and DelphesMA5tune for Apple computers with M1 chips on Mac OS v13.0.

Cheers!
